src/main/java/com/stripe/net/StripeRequest.java

```diff
@@ -71,7 +71,55 @@ public StripeRequest(
       this.options = (options != null) ? options : RequestOptions.getDefault();
       this.method = method;
       this.url = buildURL(method, url, params);
-      this.content = buildContent(method, params);
+
+      if (this.options instanceof RawRequestOptions) {
+        RawRequestOptions rawOptions = (RawRequestOptions) this.options;
+        this.content = buildContent(method, params, rawOptions.getApiMode());
+      } else {
+        this.content = buildContent(method, params, RawRequestOptions.ApiMode.STANDARD);
+      }
+
+      this.headers = buildHeaders(method, this.options);
+    } catch (IOException e) {
+      throw new ApiConnectionException(
+          String.format(
+              "IOException during API request to Stripe (%s): %s "
+                  + "Please check your internet connection and try again. If this problem persists,"
+                  + "you should check Stripe's service status at https://twitter.com/stripestatus,"
+                  + " or let us know at support@stripe.com.",
+              Stripe.getApiBase(), e.getMessage()),
+          e);
+    }
+  }
+
+  /**
+   * Initializes a new instance of the {@link StripeRequest} class.
+   *
+   * @param method the HTTP method
+   * @param url the URL of the request
+   * @param content the body of the request
+   * @param options the special modifiers of the request
+   * @throws StripeException if the request cannot be initialized for any reason
+   */
+  public StripeRequest(
+      ApiResource.RequestMethod method,
+      String url,
+      String content,
+      RequestOptions options)
+      throws StripeException {
```

Comment:
This overload might be a breaking change. If for some reason a user were importing `StripeRequest` and passing in `null` for params, then they'd get an ambiguous method error

Reply:
Should we instead do an overload that just accepts an additional string argument `content`? And call that with a `null` params for `rawRequest`?

Reply:
we can do a static factory function like in .net https://github.com/stripe/stripe-dotnet/pull/2690/files#diff-ac904c400fdd4a5fb6b51487f1d8b4f8b5cc1076c67c44b590c9b0ce53a5ecc3R116